### PR TITLE
Fix NullPointer Exceptions when using old mapper api

### PR DESCRIPTION
**Pull request by lshmouse**
We encountered several null pointer exceptions when running LR_test using angel-submit.
Just fix these some problems.

```
2017-06-27 15:21:17,257 ERROR [main] com.tencent.angel.master.AngelApplicationMaster: Split failed, error: java.lang.ClassCastException: org.apache.hadoop.mapreduce.lib.input.CombineTextInputFormat cannot be cast to org.apache.hadoop.mapred.InputFormat
2017-06-27 15:21:17,257 FATAL [main] com.tencent.angel.master.AngelApplicationMaster: Error starting AppMaster
java.io.IOException: java.lang.ClassCastException: org.apache.hadoop.mapreduce.lib.input.CombineTextInputFormat cannot be cast to org.apache.hadoop.mapred.InputFormat
  at com.tencent.angel.master.AngelApplicationMaster.recoveryDataSplits(AngelApplicationMaster.java:817)
  at com.tencent.angel.master.AngelApplicationMaster.initAndStart(AngelApplicationMaster.java:693)   
  at com.tencent.angel.master.AngelApplicationMaster$1.run(AngelApplicationMaster.java:581)          
  at java.security.AccessController.doPrivileged(Native Method)                                      
  at javax.security.auth.Subject.doAs(Subject.java:422)                                              
  at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1885)            
  at com.tencent.angel.master.AngelApplicationMaster.main(AngelApplicationMaster.java:578)           
Caused by: java.lang.ClassCastException: org.apache.hadoop.mapreduce.lib.input.CombineTextInputFormat cannot be cast to org.apache.hadoop.mapred.InputFormat
  at com.tencent.angel.master.data.DataSpliter.generateSplitsUseOldAPI(DataSpliter.java:175)         
  at com.tencent.angel.master.data.DataSpliter.generateSplits(DataSpliter.java:94)                   
  at com.tencent.angel.master.AngelApplicationMaster.recoveryDataSplits(AngelApplicationMaster.java:813)
  ... 6 more
```
```
2017-06-27 15:53:07,921 ERROR [pool-8-thread-1] com.tencent.angel.worker.storage.DFSStorageOldAPI: init reader error 
java.lang.NullPointerException                                                                          
  at org.apache.hadoop.mapred.lib.CombineFileRecordReader.initNextRecordReader(CombineFileRecordReader.java:143)
  at org.apache.hadoop.mapred.lib.CombineFileRecordReader.<init>(CombineFileRecordReader.java:122)   
  at org.apache.hadoop.mapred.lib.CombineTextInputFormat.getRecordReader(CombineTextInputFormat.java:47)
  at com.tencent.angel.worker.storage.DFSStorageOldAPI.initReader(DFSStorageOldAPI.java:63)             
  at com.tencent.angel.worker.storage.DataBlockManager.getReader(DataBlockManager.java:96)              
  at com.tencent.angel.worker.task.TaskContext.getReader(TaskContext.java:106)                          
  at com.tencent.angel.ml.classification.lr.LRTrainTask.preProcess(LRTrainTask.scala:80)                
  at com.tencent.angel.worker.task.Task.runUser(Task.java:92)                                           
  at com.tencent.angel.worker.task.Task.run(Task.java:70)                                               
  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)                    
  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)                    
  at java.lang.Thread.run(Thread.java:745)
```

### CODE REVIEW

**Comment by lshmouse** [comment]
The reason why we use old mapper api is that we encountered hdfs token delegation probem when using the new api.

```
2017-06-27 17:41:41,010 FATAL [main] com.tencent.angel.master.AngelApplicationMaster: Error starting AppMaster
java.io.IOException: org.apache.hadoop.ipc.RemoteException(java.io.IOException): Delegation Token can be issued only with kerberos or web authentication
  at org.apache.hadoop.hdfs.server.namenode.FSNamesystem.getDelegationToken(FSNamesystem.java:6512)  
  at org.apache.hadoop.hdfs.server.namenode.NameNodeRpcServer.getDelegationToken(NameNodeRpcServer.java:560)
  at org.apache.hadoop.hdfs.protocolPB.ClientNamenodeProtocolServerSideTranslatorPB.getDelegationToken(ClientNamenodeProtocolServerSideTranslatorPB.java:939)
  at org.apache.hadoop.hdfs.protocol.proto.ClientNamenodeProtocolProtos$ClientNamenodeProtocol$2.callBlockingMethod(ClientNamenodeProtocolProtos.java)
  at org.apache.hadoop.ipc.ProtobufRpcEngine$Server$ProtoBufRpcInvoker.call(ProtobufRpcEngine.java:585)
  at org.apache.hadoop.ipc.RPC$Server.call(RPC.java:928)                                             
  at org.apache.hadoop.ipc.Server$Handler$1.run(Server.java:2039)                                    
  at org.apache.hadoop.ipc.Server$Handler$1.run(Server.java:2035)                                    
  at java.security.AccessController.doPrivileged(Native Method)                                      
  at javax.security.auth.Subject.doAs(Subject.java:396)                                              
  at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1806)            
  at org.apache.hadoop.ipc.Server$Handler.run(Server.java:2033)                                      
                                                                                                     
  at com.tencent.angel.master.AngelApplicationMaster.recoveryDataSplits(AngelApplicationMaster.java:816)
  at com.tencent.angel.master.AngelApplicationMaster.initAndStart(AngelApplicationMaster.java:692)   
  at com.tencent.angel.master.AngelApplicationMaster$1.run(AngelApplicationMaster.java:580)          
  at java.security.AccessController.doPrivileged(Native Method)                                      
  at javax.security.auth.Subject.doAs(Subject.java:422)                                              
  at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1885)            
  at com.tencent.angel.master.AngelApplicationMaster.main(AngelApplicationMaster.java:577)           
Caused by: org.apache.hadoop.ipc.RemoteException(java.io.IOException): Delegation Token can be issued only with kerberos or web authentication
  at org.apache.hadoop.hdfs.server.namenode.FSNamesystem.getDelegationToken(FSNamesystem.java:6512)  
  at org.apache.hadoop.hdfs.server.namenode.NameNodeRpcServer.getDelegationToken(NameNodeRpcServer.java:560)
  at org.apache.hadoop.hdfs.protocolPB.ClientNamenodeProtocolServerSideTranslatorPB.getDelegationToken(ClientNamenodeProtocolServerSideTranslatorPB.java:939)
  at org.apache.hadoop.hdfs.protocol.proto.ClientNamenodeProtocolProtos$ClientNamenodeProtocol$2.callBlockingMethod(ClientNamenodeProtocolProtos.java)
  at org.apache.hadoop.ipc.ProtobufRpcEngine$Server$ProtoBufRpcInvoker.call(ProtobufRpcEngine.java:585)
  at org.apache.hadoop.ipc.RPC$Server.call(RPC.java:928)                                             
  at org.apache.hadoop.ipc.Server$Handler$1.run(Server.java:2039)                                    
  at org.apache.hadoop.ipc.Server$Handler$1.run(Server.java:2035)                                    
  at java.security.AccessController.doPrivileged(Native Method)                                      
  at javax.security.auth.Subject.doAs(Subject.java:396)                                              
  at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1806)            
  at org.apache.hadoop.ipc.Server$Handler.run(Server.java:2033)                                  
                                                                                                     
  at org.apache.hadoop.ipc.Client.call(Client.java:1477)                                             
  at org.apache.hadoop.ipc.Client.call(Client.java:1408)                                             
  at org.apache.hadoop.ipc.ProtobufRpcEngine$Invoker.invoke(ProtobufRpcEngine.java:232)              
  at com.sun.proxy.$Proxy9.getDelegationToken(Unknown Source)                                        
  at org.apache.hadoop.hdfs.protocolPB.ClientNamenodeProtocolTranslatorPB.getDelegationToken(ClientNamenodeProtocolTranslatorPB.java:988)
  at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)                                     
  at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)                   
  at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)           
  at java.lang.reflect.Method.invoke(Method.java:498)                                                
  at org.apache.hadoop.io.retry.RetryInvocationHandler.invokeMethod(RetryInvocationHandler.java:187) 
  at org.apache.hadoop.io.retry.RetryInvocationHandler.invoke(RetryInvocationHandler.java:102)       
  at com.sun.proxy.$Proxy10.getDelegationToken(Unknown Source)                                       
  at org.apache.hadoop.hdfs.DFSClient.getDelegationToken(DFSClient.java:1058)                        
  at org.apache.hadoop.hdfs.DistributedFileSystem.getDelegationToken(DistributedFileSystem.java:1433)
  at org.apache.hadoop.fs.FileSystem.collectDelegationTokens(FileSystem.java:529)                    
  at org.apache.hadoop.fs.FileSystem.addDelegationTokens(FileSystem.java:507)                        
  at org.apache.hadoop.hdfs.DistributedFileSystem.addDelegationTokens(DistributedFileSystem.java:2118)
  at org.apache.hadoop.mapreduce.security.TokenCache.obtainTokensForNamenodesInternal(TokenCache.java:140)
  at org.apache.hadoop.mapreduce.security.TokenCache.obtainTokensForNamenodesInternal(TokenCache.java:100)
  at org.apache.hadoop.mapreduce.security.TokenCache.obtainTokensForNamenodes(TokenCache.java:80)    
  at org.apache.hadoop.mapreduce.lib.input.FileInputFormat.listStatus(FileInputFormat.java:242)      
  at org.apache.hadoop.mapreduce.lib.input.CombineFileInputFormat.getSplits(CombineFileInputFormat.java:217)
  at com.tencent.angel.master.data.DataSpliter.generateSplitsUseNewAPI(DataSpliter.java:154)         
  at com.tencent.angel.master.data.DataSpliter.generateSplits(DataSpliter.java:110)                  
  at com.tencent.angel.master.AngelApplicationMaster.recoveryDataSplits(AngelApplicationMaster.java:812)
  ... 6 more
```

File: angel-ps/core/src/main/java/com/tencent/angel/conf/AngelConfiguration.java [hunk 1]
```diff
@@ -28,7 +28,7 @@
 import com.tencent.angel.worker.task.BaseTask;
 
 import org.apache.hadoop.conf.Configuration;
-import org.apache.hadoop.mapreduce.lib.input.CombineTextInputFormat;
+import org.apache.hadoop.mapred.lib.CombineTextInputFormat;
```
Comment (paynie) [inline angel-ps/core/src/main/java/com/tencent/angel/conf/AngelConfiguration.java]: Thanks for your pr. It is not a good idea to modify the default parameter value. If you want to use the old version of mapred inputformat,  you can use the following parameters like this:`-Dmapred.mapper.new-api=false -Dangel.input.format=org.apache.hadoop.mapred.lib.CombineTextInputFormat`

**Comment by lshmouse** [comment]
@paynie ClassCastException will be  threw if using the default value: 
import org.apache.hadoop.mapreduce.lib.input.CombineTextInputFormat;

```
2017-06-27 15:21:17,257 ERROR [main] com.tencent.angel.master.AngelApplicationMaster: Split failed, error: java.lang.ClassCastException: org.apache.hadoop.mapreduce.lib.input.CombineTextInputFormat cannot be cast to org.apache.hadoop.mapred.InputFormat
2017-06-27 15:21:17,257 FATAL [main] com.tencent.angel.master.AngelApplicationMaster: Error starting AppMaster
java.io.IOException: java.lang.ClassCastException: org.apache.hadoop.mapreduce.lib.input.CombineTextInputFormat cannot be cast to org.apache.hadoop.mapred.InputFormat
  at com.tencent.angel.master.AngelApplicationMaster.recoveryDataSplits(AngelApplicationMaster.java:817)
  at com.tencent.angel.master.AngelApplicationMaster.initAndStart(AngelApplicationMaster.java:693)   
  at com.tencent.angel.master.AngelApplicationMaster$1.run(AngelApplicationMaster.java:581)          
  at java.security.AccessController.doPrivileged(Native Method)                                      
  at javax.security.auth.Subject.doAs(Subject.java:422)                                              
  at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1885)            
  at com.tencent.angel.master.AngelApplicationMaster.main(AngelApplicationMaster.java:578)           
Caused by: java.lang.ClassCastException: org.apache.hadoop.mapreduce.lib.input.CombineTextInputFormat cannot be cast to org.apache.hadoop.mapred.InputFormat
  at com.tencent.angel.master.data.DataSpliter.generateSplitsUseOldAPI(DataSpliter.java:175)         
  at com.tencent.angel.master.data.DataSpliter.generateSplits(DataSpliter.java:94)                   
  at com.tencent.angel.master.AngelApplicationMaster.recoveryDataSplits(AngelApplicationMaster.java:813)
  ... 6 more
```

**Comment by lshmouse** [comment]
@paynie  Sorry, my mistake. I will revert it.

**Pull request merged**